### ISSUE TITLE
FVS: Version control for biosum_keywords.kcp.template

### ISSUE THREAD

**Issue by druepdx** [USFS-PNW/Fia-Biosum-Manager#373]
Currently, edits to the template are overwritten when FVSIn is run. @lbross's suggestion to combat this is the following:

1. Check to see if the file is already in the 'copy to' location in the validation section before the process starts to run. That way BioSum doesn't prompt multiple times for multiple variants.
2. If it is, pop a messageBox asking the user if they want to overwrite
3. If they say yes, save the old copy off with a timestamp
4. If no, then skip the copying

**Comment by jsfried** [comment]
That sounds like a great solution-- thanks!

**Comment by Cat-Ducat-USDA** [comment]
@lbross, quick question - could you clarify what you mean by " 'copy to' location in the validation section?" (in your first suggestion)

**Comment by lbross** [comment]
BioSum wants to copy the BioSum_Keywords.kcp.template to the /fvs/data folder. We want to make sure it exists there before asking the analyst if they want to overwrite the file. We also want to only ask the question once even if multiple variants are selected.

**Comment by Cat-Ducat-USDA** [comment]
Alrighty, SO: took some screenshots in the new project (but not a fresh FVSIn, I'd already made one) to validate this:

1. biosum keywords kcp template time stamp and file size PRE fvsin rewrite

![Image](https://github.com/user-attachments/assets/238bb687-d70d-47f7-87c2-cddd6d175ffb)

2. First prompt warning records already existed for those state codes blah blah

![Image](https://github.com/user-attachments/assets/b0461674-a453-4f8c-8e4e-fb39bb6e0b5b)

3. Second prompt asking user if they want to overwrite (I said yes)

![Image](https://github.com/user-attachments/assets/c03cf216-8de4-4db0-83c2-02bba0a500ef)

4. File directory w/ overwritten template and date-stamped backup -- I thought it seemed a little odd that the date stamps are the same on the new and old templates... perhaps just because I did not change the template at all?

![Image](https://github.com/user-attachments/assets/73bbc00a-bac1-4838-aed1-034064bc47cc)

**Comment by jsfried** [comment]
Well if you did not change the default biosum keywords template since 10/25 then it would not be surprised that it would have the same date stamp. Above, you say "(but not a fresh FVSIn, I'd already made one)"-- does that mean you made one using 5.11.2? If not, then it seems like you would need to have made one in 5.11.2 because the creation of that FVSIn (and what gets stuck in there for site class etc.) is possibly different for some cases than for 5.11.1 created FVSIn files, right?

**Comment by Cat-Ducat-USDA** [comment]
Yes, I'd already made an FVSIn earlier today for the 5.11.2 one cycle project I'm building. So in this case, I overwrote the one I created today.

**Comment by Cat-Ducat-USDA** [comment]
I hypothesized the one I made today would've had a date stamp of today, though.

**Comment by Cat-Ducat-USDA** [comment]
Ahhh, that makes sense. I have no further questions, can't speak for Jeremy.
